include/service/entities/site_service.js

```diff
@@ -123,6 +142,16 @@ module.exports = function SiteServiceModule(pb) {
     };
 
     /**
+     * Get all of the redirect_hosts in the database
+     * @method getRedirectHosts
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getRedirectHosts = function(cb) {
+      var dao = new pb.DAO();
+      dao.q(REDIRECT_HOSTS, { select: pb.DAO.SELECT_ALL, where: {} }, cb);
+    };
```

Comment:
This would require iterating over every active site's old host names to find which active site to redirect to. That's a tad more expensive than grabbing an active site uid from a hash array of hostnames that need to be redirected.